src/components/AccessControl/AccessControl.vue

```diff
@@ -5,19 +5,35 @@
 </template>
 
 <script>
+import userState from '../../utils/auth/state/userState';
+
 export default {
   name: 'access-control',
   props: {
-    roles: { type: Array, default: () => [] },
-    role: { type: String, default: '' },
-    componentUserId: { type: Number, default: -1 },
-    userId: { type: Number, default: -2 },
+    roles: { type: Array, required: true },
     _class: { type: Array, default: () => [] },
   },
+  data() {
+    return {
+      userRoleNum: -1,
+    };
+  },
+  created() {
+    this.userRoleNum = userState.getUserAdminLevel();
+  },
   computed: {
+    userRole() {
+      switch (this.userRoleNum) {
+        case 0:
+          return 'user';
```

Comment:
Can we use a map of code to user here with objects?

Reply:
There will be other places we reference these values.

Reply:
If we need to use that code else where in the frontend, we should use a map. However, the idea of this component (thanks @ZachOBrien ) is that you don't need to know what privilege level users are anywhere else.